algorithms/kernel/dtrees/forest/regression/df_regression_train_dense_default_impl.i

```diff
@@ -348,10 +352,10 @@ int OrderedRespHelper<algorithmFPType, cpu>::findBestSplitForFeatureSorted(algor
         if (!nFeatIdx[i]) continue;
         nLeft = (split.featureUnordered ? nFeatIdx[i] : nLeft + nFeatIdx[i]);
         if ((nLeft == n) //last split
-            || ((n - nLeft) < nMinSplitPart))
+            || ((n - nLeft) < nMinSplitPart) || ((n - nLeft) < minWeightLeaf))
```

Comment:
It affects the performance potentially. Need to check before merging.

Reply:
Yes, it'll be checked